lib/foreman/renderer.rb

```diff
@@ -269,5 +299,99 @@ def pxe_config
       @kernel = @host.operatingsystem.kernel @host.arch
       @initrd = @host.operatingsystem.initrd @host.arch
     end
+
+    # registeres host to be observed for call of deprecated methods
+    def register_method_deprecations_for(host)
+      return if !@host || !@host.respond_to?(:info) || @host.respond_to?(:monitored_for_deprecations)
+      host.class_eval { include MethodDeprecations }
```

Comment:
If using a mixin, it should be included in Host always rather than at runtime - if it introduces a behaviour change for these methods, tests won't be reliable or indicative if they're not always run either with it. If this is never included in tests, then this introduces a lot of untested code.